exonum/src/events/noise/wrappers/sodium_wrapper/handshake.rs

```diff
@@ -97,10 +97,20 @@ impl NoiseHandshake {
     pub fn finalize<S: AsyncRead + AsyncWrite + 'static>(
         self,
         stream: S,
-    ) -> Result<Framed<S, MessagesCodec>, io::Error> {
+    ) -> Result<(Framed<S, MessagesCodec>, x25519::PublicKey), io::Error> {
         let noise = self.noise.into_transport_mode()?;
+        let remote_static_key = {
+            let rs = noise.session.get_remote_static().unwrap_or_else(|| {
```

Comment:
New clippy requires `unwrap_or_else`, since it's lazy.

Reply:
I prefer `expect` either. Clippy forbids it on `warn` level: https://rust-lang-nursery.github.io/rust-clippy/master/index.html#expect_fun_call. So I think we could use `expect`.

Reply:
We discussed this earlier [here](https://github.com/exonum/exonum/pull/752#pullrequestreview-131493028) (you have to show the thread to see the discussion).
I guess we should follow the accepted convention.

Reply:
There is no function call in `expect` in this case though, it directly uses a statically allocated string. Does this still trigger clippy? I'd say this is a bug in clippy then, since the triggering code does not match the lint description.